props/websubmit/src/main.rs

```diff
@@ -85,16 +93,108 @@ impl DeletionProp {
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|node| self.cx.has_marker(marker!(stores), *node))
+                .collect::<Vec<_>>();
+            let mut sensitives = self
+                .cx
+                .all_nodes_for_ctrl(*c_id)
+                .filter(|node| self.cx.has_marker(marker!(sensitive), *node));
+
+            let controller_valid = sensitives.all(|sens| {
+                stores.iter().all(|&store| {
+                    // sensitive flows to store implies some scope flows to store callsite
+                    !(self
+                        .cx
+                        .flows_to(sens, store, paralegal_policy::EdgeType::Data))
+                        || store.typ.as_call_site().is_some_and(|cs| {
+                            // The sink that scope flows to may be another CallArgument attached to the store's CallSite, it doesn't need to be store itself.
+                            let found_scope = scopes.iter().any(|scope| {
+                                self.cx.flows_to(
+                                    *scope,
+                                    Node {
+                                        ctrl_id: store.ctrl_id,
+                                        typ: cs.into(),
+                                    },
+                                    paralegal_policy::EdgeType::Data,
+                                )
+                            });
+                            assert_error!(
+                                self.cx,
+                                found_scope,
+                                format!(
+                                    "Stored sensitive isn't scoped. sensitive {} stored here: {}",
+                                    self.cx.describe_node(sens),
+                                    self.cx.describe_node(store)
+                                )
+                            );
+                            found_scope
+                        })
+                })
+            });
+
+            assert_error!(
+                self.cx,
+                controller_valid,
+                format!(
+                    "Violation detected for controller: {}",
+                    self.cx.describe_def(*c_id)
+                ),
+            );
+        }
+    }
+}
+
+fn sc_policy(ctx: Arc<Context>) -> Result<()> {
+    ctx.named_policy(Identifier::new_intern("Scoped Storage Policy"), |ctx| {
+        ScopedStorageProp::new(ctx).check();
+        Ok(())
+    })
+}
+
 fn main() -> Result<()> {
     let ws_dir = std::env::args()
         .nth(1)
-        .ok_or_else(|| anyhow!("expected an argument"))?;
+        .ok_or_else(|| anyhow!("expected format: cargo run <path> [policy]"))?;
+    let prop_name = std::env::args().nth(2);
+
+    let prop = match prop_name {
+        Some(s) => match s.as_str() {
+            "sc" => sc_policy,
+            "del" => del_policy,
+            _ => |_| Err(anyhow!("don't recognize inputted property name")),
```

Comment:
Please make the error report also the name that it didn't recognize. Also there's a macro for this pattern called `bail!` so you could do `other => bail!("don't recognize the property name '{other}'"),`

Reply:
how fancy! done